src/main/java/bio/terra/testrunner/runner/TestRunner.java

```diff
@@ -394,6 +338,39 @@ private void executeTestConfigurationNoGuaranteedCleanup() throws Exception {
     }
   }
 
+  private VersionScript instantiateVersionScript(
+      VersionScriptSpecification versionScriptSpecification) {
+    try {
+      VersionScript versionScript =
+          versionScriptSpecification.scriptClass.getDeclaredConstructor().newInstance();
+      // set any parameters specified by the configuration
+      versionScript.setParameters(versionScriptSpecification.parameters);
+
+      // call the determineVersion method to get the version
+      logger.info("Version: Calling {}.determineVersion()", versionScript.getClass().getName());
+      return versionScript;
+    } catch (InstantiationException | IllegalAccessException e) {
+      logger.error(
+          "Version: Error calling constructor of VersionScript class: {}",
+          versionScriptSpecification.name,
+          e);
+      throw new IllegalArgumentException(
+          "Error calling constructor of VersionScript class: " + versionScriptSpecification.name,
+          e);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private VersionScriptResult determineVersion(
+      VersionScript versionScript, ServerSpecification server) {
+    try {
+      return versionScript.determineVersion(server);
```

Comment:
Similar to the previous comment, I think you don't need to wrap this in a `RuntimeException`. Can you just call them both directly in the method above? e.g.:
```
      versionScriptResults =
          versionScripts.stream()
              .map(
                  versionScript -> {
                    logger.info(
                        "Version: Calling {}.determineVersion()",
                        versionScript.getClass().getName());
                    versionScript.setParameters(versionScriptSpecification.parameters);
                    return versionScript.determineVersion(versionScript, config.server);
                  })
              .collect(Collectors.toList());
```

Reply:
Simplified code with a for-loop.